Source/themes.cpp

```diff
@@ -214,74 +214,55 @@ BOOL TFit_Obj3(int t)
 
 BOOL CheckThemeReqs(int t)
 {
-	BOOLEAN rv; // al
-	int v2;     // ecx
-	int v3;     // ecx
-	int v4;     // ecx
-	int v5;     // ecx
-	BOOLEAN v6; // zf
-	int v7;     // ecx
-	int v8;     // ecx
-	int v9;     // ecx
-
-	rv = 1;
-	if (t <= 10) {
-		if (t != 10) {
-			v2 = t - 1;
-			if (v2) {
-				v3 = v2 - 2;
-				if (v3) {
-					v4 = v3 - 2;
-					if (v4) {
-						v5 = v4 - 2;
-						if (v5) {
-							if (v5 != 2) {
-								return rv;
-							}
-							v6 = pFountainFlag == 0;
-						} else {
-							v6 = bFountainFlag == 0;
-						}
-					LABEL_21:
-						if (!v6) {
-							return rv;
-						}
-						return 0;
-					}
-				}
-			}
-			if (leveltype != 3) {
-				v6 = leveltype == DTYPE_HELL;
-				goto LABEL_21;
-			}
-			return 0;
+	BOOL rv;
+
+	rv = TRUE;
+	switch (t) {

```

Comment:
I haven't checked this enum against the PSX, but this is from our IDB:

```c
// Theme IDs.
typedef enum {
	THEME_ID_BARREL             = 0,
	THEME_ID_SHRINE             = 1,
	THEME_ID_MONSTER_PIT        = 2,
	THEME_ID_SKELETON_ROOM      = 3,
	THEME_ID_TREASURE           = 4,
	THEME_ID_LIBRARY            = 5,
	THEME_ID_TORTURE            = 6,
	THEME_ID_BLOOD_FOUNTAIN     = 7,
	THEME_ID_DECAPITATED        = 8,
	THEME_ID_PURIFYING_FOUNTAIN = 9,
	THEME_ID_ARMOR_STAND        = 10,
	THEME_ID_GOAT_SHRINE        = 11,
	THEME_ID_CAULDRON           = 12,
	THEME_ID_MURKY_FOUNTAIN     = 13,
	THEME_ID_TEAR_FOUNTAIN      = 14,
	THEME_ID_BURNING_CROSS      = 15,
	THEME_ID_WEAPON_RACK        = 16,
	THEME_ID_NONE               = 255,
} theme_id;
```

Reply:
A slightly shorter version of that enum exists in enums.h :)